src/Template/Scaffolds/index.ctp

```diff
@@ -40,5 +76,34 @@
         </table>
     </div>
 
+    <?php
+    if ($hasBulkActions) {
```

Comment:
I think, each of these bits should be an element. so the tempalte includes:

    <?= $this->element('bulk_actions/links') ?>

and the element contains:

    <?php 
    if (empty($bulkActions))( {
        return;
    }
    echo $this->....

It makes the index template easier to read/understand/manage, and the cost of using elements aught to be insignificant when crud-view is in use.